presto-iceberg/src/main/java/com/facebook/presto/iceberg/Partition.java

```diff
@@ -235,7 +235,9 @@ public static Map<Integer, Object> toMap(Map<Integer, Type.PrimitiveType> idToTy
         ImmutableMap.Builder<Integer, Object> map = ImmutableMap.builder();
         idToMetricMap.forEach((id, value) -> {
             Type.PrimitiveType type = idToTypeMapping.get(id);
-            map.put(id, Conversions.fromByteBuffer(type, value));
+            if (type != null) {
```

Comment:
Just for my edification - When can this happen? Should it be an explicit error instead of ignoring the missing type?

Reply:
This occurs when the `idToMetricMap` contains an entry for a non-primitive type, since the `idToTypeMapping` only contains mappings for primitive types (e.g. we were using a lot of functions with `row` types). I'm not sure the cases when this actually occurs, but I had to put this check in due to some NPEs when running queries with some of the sample tables.

IIRC it may be that this change might not be explicitly required for this PR. I think it got chucked in to this commit by accident. If not included here I will probably include it in another PR down the road.

Reply:
I'd suggest removing for now. And then add it later with a test case that triggers this

Reply:
fixed